geomesa-process/geomesa-process-vector/src/test/scala/org/locationtech/geomesa/process/transform/ArrowConversionProcessTest.scala

```diff
@@ -42,20 +44,30 @@ class ArrowConversionProcessTest extends Specification {
 
   "ArrowConversionProcess" should {
     "encode an empty feature collection" in {
+      ArrowAllocator.getAllocatedMemory mustEqual 0
       val bytes = process.execute(new ListFeatureCollection(sft), null, null, null, null, null, null, null, null, null).reduce(_ ++ _)
       WithClose(SimpleFeatureArrowFileReader.streaming(() => new ByteArrayInputStream(bytes))) { reader =>
         reader.sft mustEqual sft
         SelfClosingIterator(reader.features()) must beEmpty
       }
+      ArrowAllocator.getAllocatedMemory mustEqual 0
     }
 
     "encode a generic feature collection" in {
+      ArrowAllocator.getAllocatedMemory mustEqual 0
       val bytes = process.execute(collection, null, null, null, null, null, null, null, null, null).reduce(_ ++ _)
       WithClose(SimpleFeatureArrowFileReader.streaming(() => new ByteArrayInputStream(bytes))) { reader =>
         reader.sft mustEqual sft
         SelfClosingIterator(reader.features()).map(ScalaSimpleFeature.copy).toSeq must
             containTheSameElementsAs(features)
       }
+      ArrowAllocator.getAllocatedMemory mustEqual 0
+    }
+
+    "encode a generic empty feature collection with dictionary values without leaking memory" in {
+      // This returns an empty iterator.
+      process.execute(new ListFeatureCollection(sft), null, null, null, Seq("name"), null, null, null, null, null)
+      ArrowAllocator.getAllocatedMemory mustEqual 0
```

Comment:
@elahrvivaz I'm curious if you can suggest places where we could test for this kind of allocation in unit tests where queries time out....